packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -46,34 +46,38 @@ abstract class SerializationManager {
     return deserializeByClassName(jsonDecode(data));
   }
 
+  bool _isType<T1, T2>(Type t) => t == T1 || t == T2;
```

Comment:
Suggestion: Use this for all generated classes as well.

Nice new interface, would be good to use this in all generated interfaces as well and possibly remove the old `getType<..>`. This way the logic doesn't diverge.

Reply:
ye I think this make sense but I also think that part needs a huge refactor anyway and it is probably better to deal with this in another task.

Reply:
Should we add an issue for it in that case and reference back to this PR?